packages/venia-concept/src/components/MiniCart/miniCart.js

```diff
@@ -142,76 +155,81 @@ class MiniCart extends Component {
     }
 
     get productOptions() {
-        const { classes } = this.props;
-
-        const itemName = this.state.focusItem
-            ? this.state.focusItem.name
-            : null;
-        const itemPrice = this.state.focusItem
-            ? this.state.focusItem.price
-            : null;
-
-        return (
-            <div className={classes.content}>
-                <div className={classes.focusItem}>
-                    {itemName}
-                    <div className={classes.price}>${itemPrice}</div>
-                </div>
-                <div className={classes.options}>Choose a Size:</div>
-            </div>
-        );
-    }
-
-    get productConfirm() {
-        const { classes } = this.props;
-
-        return (
-            <div className={classes.save}>
-                <Button onClick={this.hideEditPanel}>Cancel</Button>
-                <Button priority="high">Update Cart</Button>
-            </div>
+        const { props, state, closeOptionsDrawer } = this;
+        const { updateItemInCart, cart } = props;
+        const { focusItem } = state;
+
+        if (focusItem === null) return;
+        const hasOptions = focusItem.options.length !== 0;
+
+        return hasOptions ? (
+            // `Name` is being used here because GraphQL does not allow
+            // filtering products by id, and sku is unreliable without
+            // a reference to the base product. Additionally, `url-key`
+            // cannot be used because we don't have page context in cart.
+            <Query
+                query={getProductDetailByName}
+                variables={{ name: focusItem.name, onServer: false }}
+            >
+                {({ loading, error, data }) => {
+                    if (error) return <div>Data Fetch Error</div>;
+                    if (loading) return loadingIndicator;
+
+                    const itemWithOptions = data.products.items[0];
+
+                    return (
+                        <Fragment>
+                            <CartOptions
+                                cartItem={focusItem}
+                                configItem={itemWithOptions}
+                                closeOptionsDrawer={closeOptionsDrawer}
+                                isLoading={cart.isLoading}
+                                updateCart={updateItemInCart}
+                            />
+                        </Fragment>
```

Comment:
What is the purpose of wrapping `CartOptions` in a `Fragment` here?

Reply:
Fixed in https://github.com/magento-research/pwa-studio/pull/732/commits/8473d2522ab199f4a40fe0e83f39925fd37983c7.